pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -702,17 +709,27 @@ func conditionallyConvertToIstioMtls(
 	sni string,
 	proxy *model.Proxy,
 	serviceMTLSMode authn_v1alpha1_applier.MutualTLSMode,
+	mtlsReady bool,
 ) *networking.TLSSettings {
 	if tls == nil {
-		if serviceMTLSMode != authn_v1alpha1_applier.MTLSStrict {
-			// Destination service is not in strict-mTLS mode, do nothing.
+		switch serviceMTLSMode {
+		case authn_v1alpha1_applier.MTLSPermissive:
+			// Permissive mTLS enabled and all service instance are labeled mTLSReady
+			if mtlsReady {
+				tls = &networking.TLSSettings{
+					Mode: networking.TLSSettings_ISTIO_MUTUAL,
+				}
+			} else {
+				return nil
+			}
+		case authn_v1alpha1_applier.MTLSStrict:
+			// Always use ISTIO_MUTUAL if destination service is strict-mTLS
+			tls = &networking.TLSSettings{
+				Mode: networking.TLSSettings_ISTIO_MUTUAL,
+			}
+		case authn_v1alpha1_applier.MTLSDisable, authn_v1alpha1_applier.MTLSUnknown:
```

Comment:
if the format is unknown, then we should treat it as permissive. if its disable, then we disable